### PR TITLE
AAKBET-618: Added custom query action to test queries from the UI

### PR DESCRIPTION
**Pull request by turegjorup**

### CODE REVIEW

File: src/Controller/AdminController.php [hunk 1]
```diff
@@ -24,15 +29,84 @@
...
+     * @param NewMaterialService   $newMaterialService
      */
-    public function __construct(UserManagerInterface $userManager)
+    public function __construct(UserManagerInterface $userManager, NewMaterialService $newMaterialService)
     {
         $this->userManager = $userManager;
+        $this->newMaterialService = $newMaterialService;
+    }
+
+    public function queryBatchAction(array $ids)
```
Comment (cableman) [inline src/Controller/AdminController.php]: Missing phpdoc function description.

File: src/Controller/AdminController.php [hunk 2]
```diff
@@ -24,15 +29,84 @@
 class AdminController extends EasyAdminController
 {
     private $userManager;
+    private $newMaterialService;
 
     /**
      * AdminController constructor.
      *
      * @param UserManagerInterface $userManager
+     * @param NewMaterialService   $newMaterialService
```
Comment (cableman) [inline src/Controller/AdminController.php]: Missing parameter descriptions?

File: src/Controller/AdminController.php [hunk 3]
```diff
@@ -24,15 +29,84 @@
...
-    public function __construct(UserManagerInterface $userManager)
+    public function __construct(UserManagerInterface $userManager, NewMaterialService $newMaterialService)
     {
         $this->userManager = $userManager;
+        $this->newMaterialService = $newMaterialService;
+    }
+
+    public function queryBatchAction(array $ids)
+    {
+        $ids = array_map(static function ($id) {
```
Comment (cableman) [inline src/Controller/AdminController.php]: Why the static keyword?

File: src/Controller/AdminController.php [hunk 4]
```diff
@@ -24,15 +29,84 @@
...
+            return (int) $id;
+        }, $ids);
+        $searches = $this->em->getRepository(Search::class)->findBy(['id' => $ids]);
+
+        $date = new \DateTimeImmutable('7 days ago');
```
Comment (cableman) [inline src/Controller/AdminController.php]: Hardcoded "7 days ago" ?

File: src/Controller/AdminController.php [hunk 5]
```diff
@@ -24,15 +29,84 @@
...
+     *
+     * @throws GuzzleException
+     * @throws InvalidArgumentException
+     */
+    public function queryAction(): Response
+    {
+        $id = $this->request->query->get('id');
+        $search = $this->em->getRepository(Search::class)->find($id);
+
+        $date = new \DateTimeImmutable('7 days ago');
```
Comment (cableman) [inline src/Controller/AdminController.php]: Hardcoded "7 days ago" ?

File: src/Controller/AdminController.php [hunk 6]
```diff
@@ -24,15 +29,84 @@
...
+        }
+    }
+
+    /**
+     * Custom action to test CQL Query.
+     *
+     * @return Response
+     *
+     * @throws GuzzleException
+     * @throws InvalidArgumentException
```
Comment (cableman) [inline src/Controller/AdminController.php]: Missing return description and exception description.

File: src/Controller/AdminController.php [hunk 7]
```diff
@@ -24,15 +29,84 @@
...
+     */
+    public function queryAction(): Response
+    {
+        $id = $this->request->query->get('id');
+        $search = $this->em->getRepository(Search::class)->find($id);
+
+        $date = new \DateTimeImmutable('7 days ago');
+        try {
+            $result = $this->newMaterialService->getNewMaterialsSinceDate($search, $date);
```
Comment (cableman) [inline src/Controller/AdminController.php]: For my info: What is the fail condition for this test? Zero results of malformed search query?

File: src/Entity/Search.php [hunk 8]
```diff
@@ -55,14 +56,15 @@ class Search
     private $category;
 
     /**
-     * @ORM\ManyToMany(targetEntity="App\Entity\Material", mappedBy="searches")
+     * @ORM\ManyToMany(targetEntity="App\Entity\Material", mappedBy="searches", fetch="EXTRA_LAZY")
```
Comment (cableman) [inline src/Entity/Search.php]: EXTRA_LAZY 👍

File: src/Entity/Search.php [hunk 9]
```diff
@@ -55,14 +56,15 @@ class Search
...
      * @Groups({"material"})
      */
     private $materials;
 
     /**
-     * @ORM\OneToMany(targetEntity="App\Entity\SearchRun", mappedBy="search", orphanRemoval=true)
+     * @ORM\OneToMany(targetEntity="App\Entity\SearchRun", mappedBy="search", orphanRemoval=true, fetch="EXTRA_LAZY")
+     * @ORM\OrderBy({"id" = "DESC"})
      */
     private $searchRuns;
```
Comment (cableman) [inline src/Entity/Search.php]: What is a searchRuns?

File: src/Service/OpenPlatform/NewMaterialService.php [hunk 10]
```diff
@@ -83,14 +82,25 @@ public function __construct(SearchService $searchService, MaterialPersistService
      *
      * @throws GuzzleException
      * @throws InvalidArgumentException
```
Comment (cableman) [inline src/Service/OpenPlatform/NewMaterialService.php]: In which case are exception throw (missing doc-block)

File: src/Service/OpenPlatform/NewMaterialService.php [hunk 11]
```diff
@@ -103,25 +113,32 @@ public function updateNewMaterialsSinceDate(Search $search, DateTimeImmutable $s
...
-            $searchRun->setIsSuccess(false);
-            $searchRun->setErrorMessage($exception->getMessage());
-        }
+    /**
+     * Get the complete CQL query thar will be preformed against OPen Search for the given Search and Date.
+     *
+     * @param Search            $search
+     * @param DateTimeImmutable $since
+     *
+     * @return string
```
Comment (cableman) [inline src/Service/OpenPlatform/NewMaterialService.php]: Missing parameter description?

File: src/Controller/AdminController.php (hunk 4 above)
Comment (turegjorup) [reply src/Controller/AdminController.php]: Should be given as a get parameter. But if we do that we should also paginate, so ignore for now

File: src/Controller/AdminController.php (hunk 5 above)
Comment (turegjorup) [reply src/Controller/AdminController.php]: Should be given as a get parameter. But if we do that we should also paginate, so ignore for now

File: src/Controller/AdminController.php (hunk 4 above)
Comment (turegjorup) [reply src/Controller/AdminController.php]: Added todo

File: src/Controller/AdminController.php (hunk 5 above)
Comment (turegjorup) [reply src/Controller/AdminController.php]: Added todo

File: src/Controller/AdminController.php (hunk 7 above)
Comment (turegjorup) [reply src/Controller/AdminController.php]: If the query is malformed or Open Search is offline

File: src/Entity/Search.php (hunk 9 above)
Comment (turegjorup) [reply src/Entity/Search.php]: Each time the Search is performed and materials saved, a "SearchRun" is registered with datetime, success, error if fail

**Pull request merged**